sdk/src/main/java/com/hedera/hashgraph/sdk/TransferTransaction.java

```diff
@@ -91,19 +59,26 @@ public TransferTransaction() {
 
     public TransferTransaction addTokenTransfer(TokenId tokenId, AccountId accountId, long value) {
         requireNotFrozen();
+        doAddTokenTransfer(getTokenTransferMap(tokenId), accountId, value);
+        return this;
+    }
 
-        // Cannot use `Map.merge()` as it uses `BiFunction`
-        var tokenTransfer = tokenTransfers.containsKey(tokenId) ?
-            Objects.requireNonNull(tokenTransfers.get(tokenId)) :
-            new HashMap<AccountId, Long>();
-        tokenTransfers.put(tokenId, tokenTransfer);
-
-        var current = tokenTransfer.containsKey(accountId) ?
-            Objects.requireNonNull(tokenTransfer.get(accountId)) :
-            0L;
+    public Map<TokenId, List<NftTransfer>> getTokenNftTransfers() {
+        return nftTransfers;
+    }
 
-        tokenTransfer.put(accountId, current + value);
+    public TransferTransaction addTokenNftTransfer(NftId nftId, AccountId sender, AccountId receiver) {
```

Comment:
Perhaps `addNftTransfer()`? I know we added `Token*` for all the classes, but I don't fee it's necessary here.